### PR TITLE
Sev-4480/fix sms subscription w logging

### PR DESCRIPTION
**Pull request by pmunin**
Continuation of https://github.com/segmentio/action-destinations/pull/1245

Which had to be reverted from production because of Error Code containing the entire message, which increased Datadog tag cardinality

This PR has a minor fix for that error code and only put there the code from Twilio response or if it's not available then statusCode

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)

### CODE REVIEW

File: packages/destination-actions/src/destinations/engage-messaging-twilio/sendSms/sms-sender.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export class SmsMessageSender extends MessageSender<Payload> {
       this.tags.push('reason:get_content_template')
       this.statsClient?.incr('actions-personas-messaging-twilio.error', 1, this.tags)
       this.logError(
-        `SMS failed request to fetch content template from Twilio Content API - ${this.settings.spaceId}, ${JSON.stringify(error)})}`
+        `SMS failed request to fetch content template from Twilio Content API - ${this.settings.spaceId} - ${error}, ${JSON.stringify(error)})}`
```
Comment (varadarajan-tw) [inline packages/destination-actions/src/destinations/engage-messaging-twilio/sendSms/sms-sender.ts:148]: Is this change intentional? Because JSON.stringify(error) will get you the information needed in the logs. Adding error object might make the long look unreadable.

**Review by varadarajan-tw — APPROVED** [review]
Added one question about the change to log statement. Otherwise, the PR LGTM.

File: packages/destination-actions/src/destinations/engage-messaging-twilio/sendSms/sms-sender.ts (hunk 1 above)
Comment (pmunin) [reply packages/destination-actions/src/destinations/engage-messaging-twilio/sendSms/sms-sender.ts:148]: Yeah, unfortunately it's not always that easy:
image
It depends on the error I guess, but I added that just in case JSON won't work for that error

Comment (varadarajan-tw) [reply packages/destination-actions/src/destinations/engage-messaging-twilio/sendSms/sms-sender.ts:148]: Got it! 👍

**Comment by marinhero** [comment]
I'm failing to understand how this is going to fix the issue as you are passing the same parameters or more. For example, I don't think we should log stuff like this `${this.settings.spaceId}` that would effectively create a log aggregate per space/source. You should be able to already filter by that within the Datadog dashboard.
